src/vs/workbench/services/editor/common/editorOpenWith.ts

```diff
@@ -155,7 +155,21 @@ export function getAllAvailableEditors(
 
 					const fileEditorInput = editorService.createEditorInput({ resource, forceFile: true });
 					const textOptions: IEditorOptions | ITextEditorOptions = options ? { ...options, override: false } : { override: false };
-					return { override: editorService.openEditor(fileEditorInput, textOptions, group) };
+					return {
+						override: (async () => {
+							// Try to replace existing editors for resource
+							const existingEditor = group.editors.find(editor => editor.resource && isEqual(editor.resource, resource));
```

Comment:
This helper method to find an editor by `resource` should ideally live in the `EditorGroup` imho. That is the best place to do the comparison of resources I think. I found more places where we would benefit from such a method ([1], [2]).

There is already a bug here where you compare the "original URI" with the "canonical URI" of the input. If any, you need to compare with `fileEditorInput.resource` because that is the canonical one.


**[1]** https://github.com/microsoft/vscode/blob/07bd0e3c351dab4f30c6b6b43ff6f0a5db0ae6ff/src/vs/workbench/contrib/notebook/browser/notebook.contribution.ts#L321-L321

**[2]** https://github.com/microsoft/vscode/blob/f0bb23ca02f320816219017234e412caa425851e/src/vs/workbench/contrib/customEditor/browser/customEditors.ts#L549-L549

Reply:
Hm, maybe a better place is inside `IEditorService` itself because the other one does not really talk about `URI`, rather typed editor inputs. We have a similar method `isOpen` that allows to pass in a resource editor input which I think is very similar:

https://github.com/microsoft/vscode/blob/66f5e9294e3af365bc62260839fe51806eb83ac5/src/vs/workbench/services/editor/browser/editorService.ts#L743-L755

We could have a new method `find` that allows to pass in editor input or resource editor input and does something similar?

Reply:
I just a change that adds a `IEditorService.getEditorsForResource` method. Let me know if this matches what you were thinking or if we should try something different